ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,193 @@
+"""
+High level processing of showers.
+
+This processor will be able to process a shower/event in 3 steps:
+- shower geometry
+- estimation of energy (optional, currently unavailable)
+- estimation of classification (optional, currently unavailable)
+
+"""
+import numpy as np
+
+from astropy.coordinates import SkyCoord, AltAz
+
+from ctapipe.core import Component, QualityQuery
+from ctapipe.core.traits import List
+from ctapipe.containers import ArrayEventContainer, ReconstructedShowerContainer
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.reco import HillasReconstructor
+
+
+DEFAULT_SHOWER_PARAMETERS = ReconstructedShowerContainer()
+
+
+class ShowerQualityQuery(QualityQuery):
+    """Configuring shower-wise data checks."""
+
+    quality_criteria = List(
+        default_value=[
+            ("Finite charge", "lambda hillas: hillas.intensity > 0"),
+            ("Finite width", "lambda hillas: hillas.width > 0"),
```

Comment:
Doing `hilas.width.value > 0` is much faster than `hillas.width > 0`.

Reply:
done (also for the other cuts)